### PR TITLE
Migrate Figure.colorbar tests to use dvc

### PR DESCRIPTION
**Pull request by willschlitzer**
As discussed in #1131, this moves the reference figures in `fig.colorbar` to use `dvc` and replaces the tests using `@check_figures_equal`.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by willschlitzer** [comment]
@GenericMappingTools/python `test_colorbar.py` originally had 18 tests, which I've shortened so far to 15 tests. Much like `fig.coast` in #1103, this is a pretty straightforward module, so I'm assuming I can remove tests further. My thought is to leave the three tests that pass different types of parameters for `shading=`, the ones that pass arguments to `scale=` and `truncate=`, and combine the tests that use variants of arguments passed to `box=` and `position=`. Thoughts?

**Comment by weiji14** [comment]
> 1. Add basemaps?
> 
> image
> 
> Do you think we should add a basemap to these tests, so that at least we know the colorbar is plotted at the correct positions.

Nah, I think it's fine as it is.

> 2. If I understand it correctly, calling `Figure.colorbar()` without any arguments also works. I think we should also test it. I think most people usually call `Figure.colorbar()` first and add more arguments to change the appearance.

Yes, but is it important enough to test this? Also seems beyond the scope of this PR.

> 3. 
> 
> image
> 
> Colorbars in two tests have messy annotations. Perhaps we should add `frame=True` to `Figure.colorbar()`?

Yes, probably good to change the annotation interval so the axis labels are visible.

File: pygmt/tests/test_colorbar.py [hunk 1]
```diff
@@ -156,61 +45,23 @@ def test_colorbar_scaled_z_values():
...
-    # Use single-character arguments for the reference image
-    fig_ref.basemap(R="0/10/0/10", J="X15c", B="a")
-    fig_ref.colorbar(C="geo", I="-0.7/0.2")
-
-    fig_test.basemap(region=[0, 10, 0, 10], projection="X15c", frame="a")
-    fig_test.colorbar(cmap="geo", shading="-0.7/0.2")
-    return fig_ref, fig_test
+    fig = Figure()
+    fig.basemap(region=[0, 10, 0, 10], projection="X15c", frame="a")
+    fig.colorbar(cmap="geo", shading=True)
```
Comment (seisman) [inline pygmt/tests/test_colorbar.py:55]:
```suggestion
    fig.colorbar(cmap="geo", shading=True, frame=True)
```

File: pygmt/tests/test_colorbar.py [hunk 2]
```diff
@@ -156,61 +45,23 @@ def test_colorbar_scaled_z_values():
...
-    # Use single-character arguments for the reference image
-    fig_ref.basemap(R="0/10/0/10", J="X15c", B="a")
-    fig_ref.colorbar(C="geo", I="-0.7/0.2")
-
-    fig_test.basemap(region=[0, 10, 0, 10], projection="X15c", frame="a")
-    fig_test.colorbar(cmap="geo", shading=[-0.7, 0.2])
-    return fig_ref, fig_test
+    fig = Figure()
+    fig.basemap(region=[0, 10, 0, 10], projection="X15c", frame="a")
+    fig.colorbar(cmap="geo", shading=[-0.7, 0.2])
```
Comment (seisman) [inline pygmt/tests/test_colorbar.py:66]:
```suggestion
    fig.colorbar(cmap="geo", shading=[-0.7, 0.2], frame=True)
```

**Review by seisman — APPROVED** [review]

**Review by seisman — APPROVED** [review]

**Pull request merged**